mlir/include/mlir/Dialect/SCF/IR/SCFOps.td

```diff
@@ -512,20 +512,23 @@ def ForallOp : SCF_Op<"forall", [
   let extraClassDeclaration = [{
     // Get lower bounds as OpFoldResult.
     SmallVector<OpFoldResult> getMixedLowerBound() {
-      Builder b(getOperation()->getContext());
-      return getMixedValues(getStaticLowerBound(), getDynamicLowerBound(), b);
+      auto maybeLowerBounds = getLowerBounds();
+      assert(maybeLowerBounds.has_value() && "expected values");
```

Comment:
I am not sure `assert` is a good idea. This some time leads to spurious failures. What is reason where it wont have a value.

Reply:
there should be no reason, which is why i made it an assert. just a bug catcher in case other things change